x-pack/plugins/alerting_builtins/server/alert_types/index_threshold/lib/core_query_types.ts

```diff
@@ -94,15 +92,27 @@ function validateAggType(aggType: string): string | undefined {
   });
 }
 
-export function validateDuration(duration: string): string | undefined {
-  try {
-    parseDuration(duration);
-  } catch (err) {
-    return i18n.translate('xpack.alertingBuiltins.indexThreshold.invalidDurationErrorMessage', {
-      defaultMessage: 'invalid duration: "{duration}"',
+export function validateGroupBy(groupBy: string): string | undefined {
+  if (groupBy === 'all' || groupBy === 'top') return;
+  return i18n.translate('xpack.alertingBuiltins.indexThreshold.invalidGroupByErrorMessage', {
+    defaultMessage: 'invalid groupBy: "{groupBy}"',
+    values: {
+      groupBy,
+    },
+  });
```

Comment:
nit.
I find early returns like this are easy to miss and misread.
Would it be possible to flip it so that the invalid group is only returned when `groupBy !== 'all' && groupBy !== 'top'` and then the function's implicit `void` return would mean it's valid?

Just feels cleaner to me.

Reply:
Or better yet, use the type? Something like this should work:

```
schema.oneOf([
    schema.literal('all'),
    schema.literal('top'),
  ])
```

Reply:
I did use `schema.oneOf([schema.literal('foo'), ...])` at one point for things like this.   But I changed back to strings with a custom validation, because the `schema` way produced an error message with submessages for each valid literal (eg, "it wasn't a foo; and it wasn't a bar; and it wasn't a ...", but even longer).  Heh.  I'll open an issue on kbn-schema for that, I think there's probably a way to fix that ...

Reply:
I opted to make the early return more obvious; put it in a block, with an empty line behind it.I _like_ early returns, but you're right, sometimes they aren't obvious ...